react/components/utils/useDeviceHook.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { useMedia } from 'use-media'
+
+export enum Device {
+  phone = 'phone',
+  tablet = 'tablet',
+  desktop = 'desktop',
+}
+
+const withDevice = Component => {
+  function WithDevice(props) {
+    // Taken at https://github.com/vtex-apps/device-detector/blob/master/react/useDevice.tsx#L22
+    const isScreenMedium = useMedia({ minWidth: '40rem' })
+    const isScreenLarge = useMedia({ minWidth: '64.1rem' })
+
+    const device = isScreenLarge
+      ? Device.desktop
+      : isScreenMedium
+      ? Device.tablet
+      : Device.phone
+
+    return <Component device={device} {...props} />
+  }
+
+  return WithDevice
+}
```

Comment:
Since this is an util for every component in the styleguide I think you could have both an `useDevice` (for those using FC's) and `withDevice` (uses the previous one to support classes).

Reply:
@matheusps I did not create a hook for this since i did not need one at the time and i do not want to create utils or code that i do not need directly. But i agree that if needed in the future, it could be a good addition.